tutorials/custom_model.ipynb

```diff
@@ -0,0 +1,1037 @@
+    "BoTorch's `Model` class only requires you to define a `posterior()` method that returns a `Posterior` object, the only requirement of which is to implement an `rsample()` function for drawing posterior samples. Specifically, we can utilize the subclass [`TorchPosterior`](https://github.com/pytorch/botorch/blob/main/botorch/posteriors/torch.py) that directly wraps a [torch distribution](https://pytorch.org/docs/stable/distributions.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[KeOps] Warning : omp.h header is not in the path, disabling OpenMP. To fix this, you can set the environment\n",
+      "                  variable OMP_PATH to the location of the header before importing keopscore or pykeops,\n",
+      "                  e.g. using os.environ: import os; os.environ['OMP_PATH'] = '/path/to/omp/header'\n",
+      "[KeOps] Warning : Cuda libraries were not detected on the system or could not be loaded ; using cpu only mode\n"
+     ]
+    }
+   ],
+   "source": [
+    "from typing import Optional, Union\n",
+    "from torch import Tensor, distributions, nn\n",
+    "from botorch.acquisition.objective import PosteriorTransform\n",
+    "from botorch.models.model import Model\n",
+    "from botorch.posteriors.posterior import Posterior\n",
+    "from botorch.posteriors.torch import TorchPosterior\n",
+    "\n",
+    "\n",
+    "class ProbabilisticRegressionModel(Model):\n",
+    "    _num_outputs: int\n",
+    "\n",
+    "    def __init__(self, train_X: Tensor, train_Y: Tensor):\n",
+    "        super(ProbabilisticRegressionModel, self).__init__()\n",
+    "        self._num_outputs = train_Y.shape[-1]\n",
+    "        # Linear layer that will compute the regression output.\n",
+    "        self.linear = nn.Linear(train_X.shape[-1], self.num_outputs)\n",
+    "\n",
+    "    @property\n",
+    "    def num_outputs(self) -> int:\n",
+    "        return self._num_outputs\n",
+    "\n",
+    "    def forward(self, x: Tensor) -> distributions.Distribution:\n",
+    "        n, p = x.squeeze().shape\n",
+    "        # For now, lets suppose we have known variance 1.\n",
+    "        return distributions.StudentT(df=n - p, loc=self.linear(x), scale=1)\n",
+    "\n",
+    "    def posterior(\n",
+    "        self,\n",
+    "        X: Tensor,\n",
+    "        output_indices: Optional[list[int]] = None,\n",
+    "        observation_noise: Union[bool, Tensor] = False,\n",
+    "        posterior_transform: Optional[PosteriorTransform] = None,\n",
+    "    ) -> Posterior:\n",
+    "        if output_indices:\n",
+    "            X = X[..., output_indices]\n",
+    "        # TorchPosterior directly wraps our torch.distributions.Distribution output.\n",
+    "        posterior = TorchPosterior(distribution=self(X))\n",
+    "        if posterior_transform:\n",
```

Comment:
Nit: Can't hurt to be explicit, although it's not likely to be problematic in this case

```suggestion
    "        if posterior_transform is not None:\n",
```